### PR TITLE
Bug 1694867:  text/extended/builds: Fix buildah in image build

### PR DESCRIPTION
**Pull request by wewang58**
Case: OCP-23721
@openshift/devexp-qe pls help to review it
/assign @coreydaley @adambkaplan 
Thanks all

### CODE REVIEW

**Comment by wewang58** [comment]
go-bindata cannot deal soft  link file, so  there's no files like: symlink  in /tmp/xxxxx/test/extended/testdata/builds/dockerignore1 when do build, then result in e2e-aws-builds failed with no symlink -> subdir, @coreydaley @adambkaplan  do you have idea about it?

**Comment by adambkaplan** [comment]
@wewang58 I wouldn't use go-bindata if you're fooling around with symlinks.

We have an existing suite for `start-build` that tests if we're processing symlinks correctly [1]. I would extend the suite by adding another `g.It(...)` context that tests the failure modes you are expecting. Use the functions to create a local git repo & add files to your liking.

[1] https://github.com/openshift/origin/blob/master/test/extended/builds/start.go#L480-L512

**Comment by wewang58** [comment]
@adambkaplan Since there's symlink test check, so just added bug 1694867 scenario to test, pls check it, you can tell me or modify it, if I am not do enough, thanks.

File: test/extended/builds/dockerfile.go [hunk 1]
```diff
@@ -119,6 +122,19 @@ USER 1001
...
+
+			g.It("testing build image with invalid dockerfile content", func() {
+				// https://bugzilla.redhat.com/show_bug.cgi?id=1694867
+				g.By("creating a build")
+				err := oc.Run("new-build").Args("--binary", "--strategy=docker", "--name=centos").Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				br, err := exutil.StartBuildAndWait(oc, "centos", fmt.Sprintf("--from-dir=%s", dockerfileAdd))
+				br.AssertFailure()
+				g.By("build log should have error about no file or directory")
+				logs, _ := br.Logs()
```
Comment (adambkaplan) [inline test/extended/builds/dockerfile.go]: iirc `br.Logs()` only returns an error if we can't get the build logs. Add another check to ensure an error doesn't occur here.

Comment (wewang58) [reply test/extended/builds/dockerfile.go]: Yes, you are right, added error check,thanks for your advice.

**Comment by adambkaplan** [comment]
/approve

/retest

Looks like there are a few unrelated flakes

**Comment by wewang58** [comment]
@adambkaplan Could you help to merge it, thanks.

**Pull request merged**